src/broker/index.js

```diff
@@ -103,35 +110,45 @@ module.exports = class Broker {
       }
 
       this.lookupRequest = lookup(this.versions)
+      await this.authenticate()
+    } finally {
+      await this.lock.release()
+    }
+  }
 
-      if (this.supportAuthenticationProtocol === null) {
-        try {
-          this.lookupRequest(apiKeys.SaslAuthenticate, requests.SaslAuthenticate)
-          this.supportAuthenticationProtocol = true
-        } catch (_) {
-          this.supportAuthenticationProtocol = false
-        }
-
-        this.logger.debug(`Verified support for SaslAuthenticate`, {
-          broker: this.brokerAddress,
-          supportAuthenticationProtocol: this.supportAuthenticationProtocol,
-        })
+  /**
+   * @private
+   * @returns {Promise}
+   */
+  async authenticate() {
```

Comment:
If this can be called from other places than the connect method, then this also needs to also be within a lock so that you don't get multiple authentication processes going on simultaneously towards the same broker.

Reply:
good point, I will add it